src/analyze/analyze-condition.c

```diff
@@ -0,0 +1,156 @@
+        {"AssertPathExistsGlob",            config_parse_unit_condition_path,   CONDITION_PATH_EXISTS_GLOB},
+        {"AssertPathIsDirectory",           config_parse_unit_condition_path,   CONDITION_PATH_IS_DIRECTORY},
+        {"AssertPathIsSymbolicLink",        config_parse_unit_condition_path,   CONDITION_PATH_IS_SYMBOLIC_LINK},
+        {"AssertPathIsMountPoint",          config_parse_unit_condition_path,   CONDITION_PATH_IS_MOUNT_POINT},
+        {"AssertPathIsReadWrite",           config_parse_unit_condition_path,   CONDITION_PATH_IS_READ_WRITE},
+        {"AssertDirectoryNotEmpty",         config_parse_unit_condition_path,   CONDITION_DIRECTORY_NOT_EMPTY},
+        {"AssertFileNotEmpty",              config_parse_unit_condition_path,   CONDITION_FILE_NOT_EMPTY},
+        {"AssertFileIsExecutable",          config_parse_unit_condition_path,   CONDITION_FILE_IS_EXECUTABLE},
+        {"AssertNeedsUpdate",               config_parse_unit_condition_path,   CONDITION_NEEDS_UPDATE},
+        {"AssertFirstBoot",                 config_parse_unit_condition_string, CONDITION_FIRST_BOOT},
+        {"AssertKernelCommandLine",         config_parse_unit_condition_string, CONDITION_KERNEL_COMMAND_LINE},
+        {"AssertKernelVersion",             config_parse_unit_condition_string, CONDITION_KERNEL_VERSION},
+        {"AssertArchitecture",              config_parse_unit_condition_string, CONDITION_ARCHITECTURE},
+        {"AssertVirtualization",            config_parse_unit_condition_string, CONDITION_VIRTUALIZATION},
+        {"AssertSecurity",                  config_parse_unit_condition_string, CONDITION_SECURITY},
+        {"AssertCapability",                config_parse_unit_condition_string, CONDITION_CAPABILITY},
+        {"AssertHost",                      config_parse_unit_condition_string, CONDITION_HOST},
+        {"AssertACPower",                   config_parse_unit_condition_string, CONDITION_AC_POWER},
+        {"AssertUser",                      config_parse_unit_condition_string, CONDITION_USER},
+        {"AssertGroup",                     config_parse_unit_condition_string, CONDITION_GROUP},
+        {"AssertControlGroupController",    config_parse_unit_condition_string, CONDITION_CONTROL_GROUP_CONTROLLER},
+
+        /* deprecated, but we should still parse them */
+        {"ConditionNull",                   config_parse_unit_condition_null,   0},
+        {"AssertNull",                      config_parse_unit_condition_null,   0},
+};
+
+static int parse_condition(Unit *u, const char *line) {
+        const char *p;
+        Condition **target;
+
+        if ((p = startswith(line, "Condition")))
+                target = &u->conditions;
+        else if ((p = startswith(line, "Assert")))
+                target = &u->asserts;
+        else
+                return log_error_errno(SYNTHETIC_ERRNO(EINVAL), "Cannot parse \"%s\".", line);
+
+        for (unsigned i = 0; i < ELEMENTSOF(condition_definitions); i++) {
```

Comment:
hmm, the ELEMENTSOF() macro returns a size_t entity, and hence I wouldn't cast this needlessly to `unsigned` here, and instead just size `size_t i` as iterator